src/Exception/InvalidArgumentException.php

```diff
@@ -7,17 +7,26 @@
 use InvalidArgumentException as CoreInvalidArgumentException;
 use Money\Exception;
 
-final class InvalidArgumentException extends CoreInvalidArgumentException implements Exception
+/**
+ * @internal
+ */
+class InvalidArgumentException extends CoreInvalidArgumentException implements Exception
 {
     /** @phpstan-pure */
-    public static function divisionByZero(): self
+    public static function divisionByZero(): DivisionByZeroException
     {
-        return new self('Cannot compute division with a zero divisor');
+        return new DivisionByZeroException('Cannot compute division with a zero divisor');
     }
 
     /** @phpstan-pure */
-    public static function moduloByZero(): self
+    public static function moduloByZero(): ModuleByZeroException
```

Comment:
Careful: "module" and "modulo" are not the same!

Reply:
handled in b91e101118de04390056431090778134c62cf64c

Reply:
Looking better now, no objections from my side!